extensions/postgres/non_transactional.go

```diff
@@ -23,7 +23,7 @@ import (
 const selectLatestExecutionQuery = `SELECT
 	ce.process_execution_id, e.is_current, e.status, e.start_time, e.timeout_seconds, e.history_event_id_sequence, e.state_execution_sequence_maps, e.info
 	FROM xdb_sys_latest_process_executions ce
-	INNER JOIN xdb_sys_process_executions e ON e.process_id = ce.process_id
+	INNER JOIN xdb_sys_process_executions e ON e.process_id = ce.process_id AND e.id = ce.process_execution_id
```

Comment:
@duoertai we forgot to join on this id which caused the describeAPI returned wrong results for id reuse policy tests